### DISCUSSION TITLE
The element doesn't respond the click without use 'wait('')'

### DISCUSSION THREAD

**Discussion by ana-sofist** [cypress-io/cypress#16040] [Questions and Help]
Hey guys, I have an issue interacting with elements and I hope you can help me.

When I try to click on a specific modal element, without some 'wait' or 'should('be.visible')', the element doesn't respond to the action. It seems like the cypress checked the element and identified the element is ready for interaction, but the dom doesn't finish his action yet, so the cypress click on the element, but there is no response.
When I force checking if is visible, using '.should('be.visible')' on element and 'cy.wait(500)' the action happens and the click works.
I would like to know what could be happening and if there is another way to resolve this using better practices.

**Comment by bahmutov** [comment]
This is probably due to the application still starting up, and the button or element is visible, yet the event listeners have not been attached yet. Read the blog post https://www.cypress.io/blog/2018/02/05/when-can-the-test-start/ and the series of the blog posts about flake: https://www.cypress.io/blog/tag/flake/

My advice: see if there is something obvious that "signals" that the modal is ready to receive clicks. Maybe there is a class set there by the app once everything is ready? If not, then just `cy.wait(500)` I am afraid, since Cypress has no way to tell that the button is NOT ready to receive clicks if the button is there, visible to the user.
